palaeontographica-abteilung-b-palaeobotany-palaeophytology.csl

```diff
@@ -0,0 +1,255 @@
+      </choose>
+    </group>
+  </macro>
+  <macro name="publisher">
+    <group delimiter=" ">
+      <choose>
+        <if type="speech paper-conference" match="none">
+          <choose>
+            <if type="book chapter" match="any">
+              <text variable="genre" suffix=": " text-case="capitalize-first"/>
+              <group delimiter=" ">
+                <text variable="publisher" prefix="(" suffix=")"/>
+                <text variable="publisher-place"/>
+              </group>
+            </if>
+            <else-if type="thesis" match="any">
+              <group delimiter=", ">
+                <text variable="genre" text-case="capitalize-first"/>
+                <text variable="publisher"/>
+              </group>
+            </else-if>
+          </choose>
+        </if>
+      </choose>
+    </group>
+    <choose>
+      <if type="speech paper-conference" match="any">
+        <text variable="event" prefix=" " font-style="italic"/>
+        <text variable="event-place" prefix=" "/>
+        <text variable="publisher" prefix=" (" suffix=")"/>
+      </if>
+    </choose>
+  </macro>
+  <macro name="issued">
+    <choose>
+      <if variable="issued">
+        <group>
+          <date variable="issued">
+            <date-part name="year"/>
+          </date>
+        </group>
+      </if>
+      <else>
+        <text term="no date"/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="issued-year">
+    <choose>
+      <if variable="issued">
+        <date variable="issued">
+          <date-part name="year"/>
+        </date>
+      </if>
+      <else>
+        <text term="no date" form="short"/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="title">
+    <group delimiter=". ">
+      <text variable="title"/>
+      <choose>
+        <if type="book">
+          <text macro="edition"/>
+        </if>
+      </choose>
+    </group>
+  </macro>
+  <macro name="container-title">
+    <choose>
+      <if type="chapter paper-conference" match="any">
+        <group delimiter=". ">
+          <text variable="container-title"/>
+          <text macro="edition"/>
+        </group>
+      </if>
+    </choose>
+  </macro>
+  <macro name="edition">
+    <choose>
+      <if type="bill book chapter graphic legal_case legislation motion_picture paper-conference report song" match="any">
+        <choose>
+          <if is-numeric="edition">
+            <number variable="edition" form="long-ordinal" text-case="capitalize-first"/>
+            <text term="edition" form="long" prefix=" " text-case="capitalize-first"/>
+          </if>
+          <else>
+            <text variable="edition"/>
+          </else>
+        </choose>
+      </if>
+    </choose>
+  </macro>
+  <macro name="number-of-pages">
+    <choose>
+      <if type="book report" match="any">
+        <group delimiter=" ">
+          <label plural="always" strip-periods="false" variable="page" form="short"/>
+          <text variable="number-of-pages"/>
```

Comment:
Given these odd instructions:

> TAYLOR, T. N. (1981): Paleobotany: An introduction to fossil plant biology. – pp. 1–589 (McGraw-Hill Book Co.) New York.
> 
> FISCHER, O. (1990): Blätterfloren aus mitteleozänen Sedimenten des südlichen Weißelster-Beckens (Profen und Scheiplitz). – pp. 1–142, unpublished PhD thesis, Humboldt University Berlin.

this should a) include thesis in line 192 and b) have `prefix="1&#8211;"` right?